consoles/sshXtermIPMI.pm

```diff
@@ -31,11 +32,40 @@ sub activate {
 
     my $testapi_console = $self->{testapi_console};
 
-    my @command = $self->backend->ipmi_cmdline;
-    push(@command, qw(sol activate));
-    my $serial = $self->{args}->{serial};
+    my $cstr              = "";
+    my @command_reset_tmp = $self->backend->ipmi_cmdline;
+    push(@command_reset_tmp, qw(mc reset cold));
+    my $command_reset = join(' ', @command_reset_tmp);
+
+    my @command_sleep_tmp = "sleep";
+    push(@command_sleep_tmp, qw(15));
+    my $command_sleep = join(' ', @command_sleep_tmp);
+
+    my $bmcipaddr = get_var('BMC_IP');
+    if ($bmcipaddr eq '') {
+        die "BMC IP address is not defined.";
+    }
+    my @command_ping_tmp = "(while true; do /usr/bin/ping -c1";
```

Comment:
What if the ping never succeeds? How about using a for loop to abort after a certain amount of retries

Reply:
Will improve.